sklearn/neighbors/binary_tree.pxi

```diff
@@ -1054,6 +1071,16 @@ cdef class BinaryTree:
         self.data_arr = np.asarray(data, dtype=DTYPE, order='C')
         self.data = get_memview_DTYPE_2D(self.data_arr)
 
+        self.new_data_arr = check_array(data, order='C', dtype=[np.float64])#, np.float32])
```

Comment:
I try to first use `self.new_data_arr` to repeat all what `self.data` does and then finally replace `self.data` with `self.new_data_arr`, so that in the development process I can still build successfully.

But I found out this is extremely confused for the reviewer. I will modify it. Sorry!